tests/test_mutations/test_collection.py

```diff
@@ -0,0 +1,145 @@
+            variable_values={"collection": newCollection},
+        )
+
+        self.assertEqual(result.errors, None)
+        self.assertEqual(
+            result.data["create_collection"]["user"]["email"], newUser["email"]
+        )
+        self.assertEqual(
+            len(result.data["create_collection"]["pictures"]), len(self.newPictures)
+        )
+        for picture in result.data["create_collection"]["pictures"]:
+            self.assertTrue(picture["picture_path"] in newCollection["pictures"])
+        self.assertEqual(
+            result.data["create_collection"]["name"], newCollection["name"]
+        )
+
+    @pytest.mark.asyncio
+    async def test_create_fail(self):
+        mutation = collection_creation_mutation
+        newUser = self.newUser
+
+        newCollection = {
+            "user": newUser["email"],
+            "name": "Best collection",
+            "pictures": [picture.picture_path for picture in self.newPictures],
+        }
+
+        newCollection2 = {
+            "user": newUser["email"],
+            "name": "Best collection",
+            "pictures": [picture.picture_path for picture in self.newPictures],
+        }
+
+        result = await schema.execute(
+            mutation,
+            variable_values={"collection": newCollection},
+        )
+
+        resultError = await schema.execute(
+            mutation,
+            variable_values={"collection": newCollection2},
+        )
+
+        self.assertEqual(result.errors, None)
+        self.assertEqual(resultError.errors, None)
+        self.assertFalse(resultError.data["create_collection"]["__typename"] is None)
+        self.assertEqual(
+            resultError.data["create_collection"]["messages"][0],
+            {
+                "field": None,
+                "kind": "VALIDATION",
+                "message": "Collection with this Name and User already exists.",
+            },
+        )
+
+    def test_add_picture(self):
+        mutation = collection_add_picture_mutation
+
+        newCollection = CollectionFactory()
+        newPicture = PictureFactory(picture_path="www.test.com")
+
+        result = schema.execute_sync(
+            mutation,
+            variable_values={
+                "collection": newCollection.id,
+                "picture": newPicture.picture_path,
+            },
+        )
+
+        self.assertEqual(result.errors, None)
+        self.assertEqual(
+            result.data["collection_add_picture"]["pictures"][0]["picture_path"],
+            newPicture.picture_path,
+        )
+
+    def test_update(self):
+        mutation = collection_update_mutation
+
+        newUser = UserFactory()
+        oldPictures = PictureFactory.create_batch(5, user=newUser)
+        newCollection = CollectionFactory(user=newUser, collection_pictures=oldPictures)
+        newPictures = PictureFactory.create_batch(10, user=newUser)
+
+        self.assertEqual(newCollection.pictures.count(), len(oldPictures))
```

Comment:
we don't need this, we can expect that the factories worked, no need to test them